src/pymor/reductors/neural_network.py

```diff
@@ -148,6 +152,21 @@ def reduce(self, hidden_layers='[(N+P)*3, (N+P)*3]', activation_function=torch.t
             # run the actual training of the neural network
             with self.logger.block(f'Performing {restarts} restarts for training ...'):
 
+                # save the best neural network when training is interrupted (if desired)
+                if self.save_training_state:
+                    import signal
+                    import sys
+
+                    # define a signal handler to call in case of a `SIGINT` signal
+                    def signal_handler(sig, frame):
+                        if hasattr(self, 'neural_network'):
+                            self.logger.warning('Aborting training, but saving best model thus far ...')
+                            self.logger.warning(f'Best neural network produces validation loss of {self.losses["val"]} ...')
+                            self.save_model()
+                        sys.exit(0)
+                    # register signal handler
+                    signal.signal(signal.SIGINT, signal_handler)
```

Comment:
I think we should consider to save and (later) restore a previously set handler here.